### PR TITLE
[main][refactor] workers deployment script

### PR DESCRIPTION
**Pull request by sora-alpaca**
## Description
Refactoring PR for deployment script on workers.

## Why?
To clean code and automatically update config file after setting and deployed.

## ChangeLogs:
- refactor cakemaxi-workers02
- refactor mdex-workers02
- refactor pancakeswap-v2-workers02
- refactor spookyswap-workers03
- refactor tomb-workers03

### CODE REVIEW

**Review by jr-alpaca — APPROVED** [review]

File: deploy/exec/workers/deploy/cakemaxi-workers02.ts [hunk 1]
```diff
@@ -263,36 +249,90 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    console.log(
-      `await timelock.executeTransaction('${workerInfos[i].WORKER_CONFIG_ADDR}', '0', 'setConfigs(address[],(bool,uint64,uint64,uint64)[])', ethers.utils.defaultAbiCoder.encode(['address[]','(bool acceptDebt,uint64 workFactor,uint64 killFactor,uint64 maxPriceDiff)[]'],[['${cakeMaxiWorker02.address}'], [{acceptDebt: true, workFactor: ${workerInfos[i].WORK_FACTOR}, killFactor: ${workerInfos[i].KILL_FACTOR}, maxPriceDiff: ${workerInfos[i].MAX_PRICE_DIFF}}]]), ${workerInfos[i].EXACT_ETA})`
-    );
-    console.log("✅ Done");
+        ],
+        workerInfos[i].EXACT_ETA,
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
+      );
+      console.log(`queue setConfigs at: ${setConfigsTx.queuedAt}`);
+      console.log("generate timelock.executeTransaction:");
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/cakemaxi-workers02.ts:257]: If use TimelockService then no need to log this out

File: deploy/exec/workers/deploy/cakemaxi-workers02.ts [hunk 2]
```diff
@@ -263,36 +249,90 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    console.log(
-      `await timelock.executeTransaction('${workerInfos[i].VAULT_CONFIG_ADDR}', '0','setWorkers(address[],address[])', ethers.utils.defaultAbiCoder.encode(['address[]','address[]'],[['${cakeMaxiWorker02.address}'], ['${workerInfos[i].WORKER_CONFIG_ADDR}']]), ${workerInfos[i].EXACT_ETA})`
-    );
-    console.log("✅ Done");
+        [[cakeMaxiWorker02.address], [workerInfos[i].WORKER_CONFIG_ADDR]],
+        workerInfos[i].EXACT_ETA,
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
+      );
+
+      console.log(`queue setWorkers at: ${setWorkersTx.queuedAt}`);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/cakemaxi-workers02.ts:295]: no need to log this out as using TimelockService already

File: deploy/exec/workers/deploy/mdex-workers02.ts [hunk 3]
```diff
@@ -288,34 +270,88 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    console.log("generate timelock.executeTransaction:");
-    console.log(
-      `await timelock.executeTransaction('${workerInfos[i].WORKER_CONFIG_ADDR}', '0', 'setConfigs(address[],(bool,uint64,uint64,uint64)[])', ethers.utils.defaultAbiCoder.encode(['address[]','(bool acceptDebt,uint64 workFactor,uint64 killFactor,uint64 maxPriceDiff)[]'],[['${mdexWorker02.address}'], [{acceptDebt: true, workFactor: ${workerInfos[i].WORK_FACTOR}, killFactor: ${workerInfos[i].KILL_FACTOR}, maxPriceDiff: ${workerInfos[i].MAX_PRICE_DIFF}}]]), ${workerInfos[i].EXACT_ETA})`
-    );
-    console.log("✅ Done");
+        ],
+        workerInfos[i].EXACT_ETA,
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
+      );
+      console.log(`queue setConfigs at: ${setConfigsTx.queuedAt}`);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/mdex-workers02.ts:277]: no need to log this out as using TimelockService already

File: deploy/exec/workers/deploy/mdex-workers02.ts [hunk 4]
```diff
@@ -288,34 +270,88 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    console.log("✅ Done");
+        [[mdexWorker02.address], [workerInfos[i].WORKER_CONFIG_ADDR]],
+        workerInfos[i].EXACT_ETA,
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
+      );
+
+      console.log(`queue setWorkers at: ${setWorkersTx.queuedAt}`);
+      console.log("generate timelock.executeTransaction:");
+      console.log(
+        `await timelock.executeTransaction('${workerInfos[i].VAULT_CONFIG_ADDR}', '0','setWorkers(address[],address[])', ethers.utils.defaultAbiCoder.encode(['address[]','address[]'],[['${mdexWorker02.address}'], ['${workerInfos[i].WORKER_CONFIG_ADDR}']]), ${workerInfos[i].EXACT_ETA})`
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/mdex-workers02.ts:319]: no need to log this out as using TimelockService already

File: deploy/exec/workers/deploy/pancakeswap-v2-workers02.ts [hunk 5]
```diff
@@ -261,36 +240,90 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    console.log("generate timelock.executeTransaction:");
-    console.log(
-      `await timelock.executeTransaction('${workerInfos[i].WORKER_CONFIG_ADDR}', '0', 'setConfigs(address[],(bool,uint64,uint64,uint64)[])', ethers.utils.defaultAbiCoder.encode(['address[]','(bool acceptDebt,uint64 workFactor,uint64 killFactor,uint64 maxPriceDiff)[]'],[['${pancakeswapV2Worker02.address}'], [{acceptDebt: true, workFactor: ${workerInfos[i].WORK_FACTOR}, killFactor: ${workerInfos[i].KILL_FACTOR}, maxPriceDiff: ${workerInfos[i].MAX_PRICE_DIFF}}]]), ${workerInfos[i].EXACT_ETA})`
-    );
-    console.log("✅ Done");
+        ],
+        workerInfos[i].EXACT_ETA,
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
+      );
+      console.log(`queue setConfigs at: ${setConfigsTx.queuedAt}`);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/pancakeswap-v2-workers02.ts:247]: no need to log this out as using TimelockService already

File: deploy/exec/workers/deploy/pancakeswap-v2-workers02.ts [hunk 6]
```diff
@@ -261,36 +240,90 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    console.log(
-      `await timelock.executeTransaction('${workerInfos[i].VAULT_CONFIG_ADDR}', '0','setWorkers(address[],address[])', ethers.utils.defaultAbiCoder.encode(['address[]','address[]'],[['${pancakeswapV2Worker02.address}'], ['${workerInfos[i].WORKER_CONFIG_ADDR}']]), ${workerInfos[i].EXACT_ETA})`
-    );
-    console.log("✅ Done");
+        [[pancakeswapV2Worker02.address], [workerInfos[i].WORKER_CONFIG_ADDR]],
+        workerInfos[i].EXACT_ETA,
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
+      );
+
+      console.log(`queue setWorkers at: ${setWorkersTx.queuedAt}`);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/pancakeswap-v2-workers02.ts:286]: no need to log this out as using TimelockService already

File: deploy/exec/workers/deploy/spookyswap-workers03.ts [hunk 7]
```diff
@@ -247,36 +223,40 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+              maxPriceDiff: workerInfos[i].MAX_PRICE_DIFF,
+            },
+          ],
+        ],
         workerInfos[i].EXACT_ETA,
-        { nonce: nonce++ }
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
       );
-      console.log(`queue setConfigs at: ${setConfigsTx.hash}`);
+      console.log(`queue setConfigs at: ${setConfigsTx.queuedAt}`);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/spookyswap-workers03.ts:253]: no need to log this out as using TimelockService already

File: deploy/exec/workers/deploy/spookyswap-workers03.ts [hunk 8]
```diff
@@ -297,24 +277,25 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-        ),
+        ["address[]", "address[]"],
+        [[spookyWorker03.address], [workerInfos[i].WORKER_CONFIG_ADDR]],
         workerInfos[i].EXACT_ETA,
-        { nonce: nonce++ }
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
       );
-      console.log(`queue setWorkers at: ${setWorkersTx.hash}`);
+
+      console.log(`queue setWorkers at: ${setWorkersTx.queuedAt}`);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/spookyswap-workers03.ts:292]: no need to log this out as using TimelockService already

File: deploy/exec/workers/deploy/tomb-workers03.ts [hunk 9]
```diff
@@ -203,101 +210,63 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+              maxPriceDiff: workerInfos[i].MAX_PRICE_DIFF,
+            },
+          ],
+        ],
         workerInfos[i].EXACT_ETA,
-        { nonce: nonce++ }
+        { gasPrice: ethers.utils.parseUnits("15", "gwei"), nonce: nonce++ }
       );
-      console.log(`queue setConfigs at: ${setConfigsTx.hash}`);
+      console.log(`queue setConfigs at: ${setConfigsTx.queuedAt}`);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/tomb-workers03.ts:263]: no need to log this out as using TimelockService already

**Review by spicysquid168 — APPROVED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**